lib/Service/MailManager.php

```diff
@@ -456,8 +457,8 @@ public function tagMessage(Account $account, string $mailbox, Message $message,
 		} catch (DoesNotExistException $e) {
 			throw new ClientException("Mailbox $mailbox does not exist", 0, $e);
 		}
-		if ($this->isPermflagsEnabled($account, $mailbox) === true) {
-			$client = $this->imapClientFactory->getClient($account);
+		$client = $this->imapClientFactory->getClient($account);
+		if ($this->isPermflagsEnabled($client, $account, $mailbox) === true) {
```

Comment:
Why not move the client back inside the if condition. It is only used inside and this would get rid of the extra `else { client.logout(); }` block.

Reply:
`ìsPermflagsEnabled` needs the client as a param now

Reply:
Ahhh, you're right.